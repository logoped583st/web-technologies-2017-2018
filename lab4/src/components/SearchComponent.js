import React from 'react';
import { connect } from 'react-redux';
import { sendRequest } from '../requsets/requests'
import { sendRequestOrgs } from '../requsets/requests'

class SearchComponent extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            text: ''
        };
        this.textChange = this.textChange.bind(this);
        this.search = this.search.bind(this);
        this.enterPress = this.enterPress.bind(this);
    }

    enterPress = (e) => {
        if (e.key === 'Enter') {
            console.log("ENTER");
            this.search();
        }
    }

    textChange(e) {
        this.setState({ text: e.target.value })
    }

    search() {
        this.props.sendRequest(this.state.text);
        this.props.sendRequestOrgs(this.state.text);
    }

    render() {
        return (
            <div className='Search'>
                <input id='inputText' type='text' onChange={this.textChange} placeholder="Username"  onKeyDown={this.enterPress}/>
                <button onClick={this.search}>Search</button>
            </div>
        );
    }
}



export default connect(null, { sendRequest, sendRequestOrgs })(SearchComponent);