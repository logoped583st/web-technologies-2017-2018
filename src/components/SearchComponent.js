import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { fetchUserRequest } from '../actions/actions'
import { fetchOrgsRequest } from '../actions/actions'
import { fetchAllReposRequest } from '../actions/actions'


class SearchComponent extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            text: '',
            textRep: '',
        };
        this.textChange = this.textChange.bind(this);
        this.search = this.search.bind(this);
        this.textChangeRepository = this.textChangeRepository.bind(this);
        // this.enterPressRepository = this.enterPressRepository.bind(this);
        // this.enterPress = this.enterPress.bind(this);

        this.searchRepository = this.searchRepository.bind(this);
    }

    // enterPress = (event) => {
    //     if (event.keyCode=== 13) {
    //         this.search();
    //     }
    // }

    // enterPressRepository = (event) => {
    //     if (event.keyCode === 13) {
    //         this.searchRepository();
    //     }
    // }

    textChange(e) {
        this.setState({ text: e.target.value });
    }

    textChangeRepository(e) {
        this.setState({ textRep: e.target.value });
    }

    search() {
        this.props.fetchUserRequest(this.state.text);
        this.props.fetchOrgsRequest(this.state.text);
    }

    searchRepository() {
        this.props.fetchAllReposRequest(this.state.textRep)
    }

    render() {
        return (
            <div>
                <div className='Search'>
                    <input id='inputText' type='text' onChange={this.textChange} placeholder="Username" onKeyDown={this.enterPress} />
                    <button  className='button'onClick={this.search}>Search User</button>
                </div>
                <div className='Search'>
                    <input id='inputText' type='text' onChange={this.textChangeRepository} placeholder="Repository Name" onKeyDown={this.enterPressRepository} />
                    <button className='button' onClick={this.searchRepository}>Search Repository</button>
                </div>
            </div>
        );
    }
}
const mapDispatchToProps = dispatch => bindActionCreators({
    fetchUserRequest,
    fetchOrgsRequest,
    fetchAllReposRequest,
}, dispatch);


export default connect(null, mapDispatchToProps)(SearchComponent);