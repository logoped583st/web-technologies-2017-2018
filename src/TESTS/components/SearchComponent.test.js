import React from 'react';

import SearchComponent  from '../../components/SearchComponent';

import { shallow, mount, configure } from 'enzyme';
import { mountToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

configure({ adapter: new Adapter() });

describe('AdditionalUserInfo component test', () => {
    const mockStore = configureStore([ thunk ]);
    const initialState = {
        
    }

    let store;

    beforeEach(() => {
        store = mockStore(initialState);
    })

    it('Search component rendered succesfully', () => {
        
        const reducerUser = mount(<SearchComponent store={store} />);
        expect(mountToJson(reducerUser)).toMatchSnapshot();

        
    });
    it('Search component rendered succesfully', () => {
        
   

        const mockLogout = jest.fn();
        const wrapper = mount(<SearchComponent startLogout={mockLogout}  store={store}/>);
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(1).simulate('click');

        wrapper.find('input').at(0).simulate('change')
        wrapper.find('input').at(1).simulate('change')

        });


});