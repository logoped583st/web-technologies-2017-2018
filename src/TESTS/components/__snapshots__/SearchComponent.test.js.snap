// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdditionalUserInfo component test Search component rendered succesfully 1`] = `
<Connect(SearchComponent)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <SearchComponent
    fetchAllReposRequest={[Function]}
    fetchOrgsRequest={[Function]}
    fetchUserRequest={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <div>
      <div
        className="Search"
      >
        <input
          id="inputText"
          onChange={[Function]}
          placeholder="Username"
          type="text"
        />
        <button
          className="button"
          onClick={[Function]}
        >
          Search User
        </button>
      </div>
      <div
        className="Search"
      >
        <input
          id="inputText"
          onChange={[Function]}
          placeholder="Repository Name"
          type="text"
        />
        <button
          className="button"
          onClick={[Function]}
        >
          Search Repository
        </button>
      </div>
    </div>
  </SearchComponent>
</Connect(SearchComponent)>
`;
